### ISSUE TITLE
C++17 does not allow dynamic exception specifications

### ISSUE THREAD

**Issue by 1opeti** [rafael2k/darkice#181]
compiling darkice-1.4 on Ubuntu 20.04 with aacplus & lame the installation fails on make because C++17 does not allow previously deprecated call to throw
Is there a work around?

**Comment by 0d0a** [comment]
see tthis to downgrade c++
https://scholtyssek.org/en/gcc-mit-c14-unterstuetzung-fuer-ubuntu-mint-installieren/

**Comment by 0d0a** [comment]
try downgrade c++ :
https://scholtyssek.org/en/gcc-mit-c14-unterstuetzung-fuer-ubuntu-mint-installieren/

**Comment by pat1** [comment]
Do not compile on Fedora 34+ and Rocky Linux too.

**Comment by 0d0a** [comment]
Darkice 1.5 succesfully compiled with lame and faac on a pi3bplus running bookworm aarch64! 🙏On 8 Dec 2023 09:55, Thierry ***@***.***> wrote:
Hi !
Did you try recently ? A PR has been submitted and already accepted.
Checkout from the dev branch and let me know

**Comment by titixbrest** [comment]
Hi @rafael2k this bug report is closed.

**Comment by 1opeti** [comment]
> Darkice 1.5 succesfully compiled with lame and faac on a pi3bplus running bookworm aarch64! 🙏On 8 Dec 2023 09:55, Thierry ***@***.***> wrote: Hi ! Did you try recently ? A PR has been submitted and already accepted. Checkout from the dev branch and let me know —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

I tried again today and it still won't compile. It may be possible to  find a version of c++ and GCC which works but so far no luck.  My option is to install some older version of ubuntu and see if that works. AAC+ is really good as a delivery platform on dodgy circuits.

**Comment by 1opeti** [comment]
I have now managed to successfully compile darkice on Ubuntu 24.04 with AAC+. I have a number of streaming servers running darkice/Icecast2 and delivering AAC+ audio as well as mp3. The HE-AAC gets through internet connections more reliably than MP3 and has a smaller footprint than MP2. I am using Barix at the receive end and they work really well with AAC+. My servers do logging and we mount remote studios on the Icecast2 server. The same server downloads news files and records audio from other stations for rebroadcast.  It makes sense to upgrade to supported versions of the server code.  It took a while but I found out how to remove the C compilers and replace them with older compilers that do not complain about unsupported exceptions. There are a couple of other edits required. I have it all in a bash script which I will figure out how to post somewhere. This is all to get access to AAC+ which for some reason is not included in the regular repository versions.
